cdp_scrapers/instances/houston.py

```diff
@@ -0,0 +1,289 @@
+            sessions=[
+                ingestion_models.Session(
+                    session_datetime=date,
+                    video_uri=main_uri + "/embed",
+                    session_index=0,
+                )
+            ],
+            event_minutes_items=self.get_event_minutes_item(agenda),
+            agenda_uri=main_uri + "/agenda",
+        )
+        return event
+
+    def get_all_elements_in_range(
+        self, time_from: datetime, time_to: datetime
+    ) -> Dict[str, Tag]:
+        """
+        Get all the meetings in a range of dates
+
+        Parameters:
+        --------------
+        time_from: datetime
+            Earliest meeting date to look at
+        time_to: datetime
+            Latest meeting date to look at
+
+        Returns:
+        --------------
+        Dict[str, Tag]
+            Dictionary of mapping between the date and the element for the meeting in
+            that date
+        """
+        if time_from.year != time_to.year:
+            raise ValueError(
+                "time_from and time_to are in different years, which is not supported"
+            )
+        date_years = dict()
+        main_url = "https://houstontx.new.swagit.com/views/408"
+        main_page = requests.get(main_url)
+        main = BeautifulSoup(main_page.content, "html.parser")
+        main_div = self.remove_extra_type(
+            main.find("div", id=self.get_diff_yearid(time_from))
+        )
+        main_table = self.remove_extra_type(main_div.find("table", id="video-table"))
+        main_tbody = self.remove_extra_type(main_table.find("tbody"))
+        main_year_elem = main_tbody.find_all("tr")
+        for year_elem in main_year_elem:
+            cells = year_elem.find_all("td")
+            date = cells[1].text.replace(",", "").strip()
+            date = datetime.strptime(date, "%b %d %Y").date()
+            if date >= time_from.date() and date <= time_to.date():
+                # date_year = [date, year_elem]
+                # date_years.append(date_year)
+                date_years[date] = year_elem
+        return date_years
+
+    def get_events(
+        self, from_dt: datetime, to_dt: datetime
+    ) -> List[ingestion_models.EventIngestionModel]:
+        """
+        Get all city council meetings information within a specific time range
+
+        Parameters:
+        --------------
+        from_dt: datetime
+            The start date of the time range
+        to_dt: datetime
+            The end date of the time range
+
+        Returns:
+        --------------
+        list[ingestion_models.EventIngestionModel]
+            A list of EventIngestionModel that contains all city council
+            meetings information within a specific time range
+        """
+        events = []
+        d = self.get_all_elements_in_range(from_dt, to_dt)
+        for date in d.keys():
+            events.append(self.get_event(date, d[date]))
```

Comment:
More pythonic to do something like this:

```Python
for date, element in d.items():
    events.append(self.get_event(date, element))
```

Also _slightly_ more efficient since this avoids additional hashing/hash table lookup.

Reply:
Thanks! Done